test/test_abort_blocks_after.py

```diff
@@ -0,0 +1,143 @@
+
+    # Block and transaction should be accepted on L2
+    contract_deploy_info = deploy(CONTRACT_PATH, inputs=["0"])
+    contract_deploy_block = get_block(parse=True)
+    assert contract_deploy_block["status"] == "ACCEPTED_ON_L2"
+    assert_tx_status(contract_deploy_info["tx_hash"], "ACCEPTED_ON_L2")
+
+    # Blocks should be aborted and transactions should be rejected
+    response = abort_blocks(contract_deploy_block["block_hash"])
+    assert response.status_code == 200
+    assert response.json()["aborted"] == [
+        contract_deploy_block["block_hash"],
+    ]
+    last_block = get_block(parse=True)
+    assert last_block["status"] == "ACCEPTED_ON_L2"
+    assert last_block["block_number"] == 0
+    contract_deploy_block_after_abort = get_block(
+        block_hash=contract_deploy_block["block_hash"], parse=True
+    )
+    assert contract_deploy_block_after_abort["status"] == "ABORTED"
+    assert_transaction(contract_deploy_info["tx_hash"], "REJECTED")
+
+    # Test RPC get block status mapping from ABORTED to REJECTED
+    rpc_aborted_block = rpc_call(
+        "starknet_getBlockWithTxs",
+        params={
+            "block_id": {"block_hash": rpc_felt(contract_deploy_block["block_hash"])}
+        },
+    )
+    assert rpc_aborted_block["result"]["status"] == "REJECTED"
+
+
+@devnet_in_background(*PREDEPLOY_ACCOUNT_CLI_ARGS)
+def test_abort_many_blocks_many_transactions():
+    """Test abort of many blocks and many transactions."""
+
+    # Block and transaction should be accepted on L2
+    contract_deploy_info = deploy(CONTRACT_PATH, inputs=["0"])
+    contract_deploy_block = get_block(parse=True)
+    assert contract_deploy_block["status"] == "ACCEPTED_ON_L2"
+    assert_tx_status(contract_deploy_info["tx_hash"], "ACCEPTED_ON_L2")
+
+    # Block and transaction should be accepted on L2
+    invoke_tx_hash = invoke(
+        calls=[(contract_deploy_info["address"], "increase_balance", [10, 20])],
+        account_address=PREDEPLOYED_ACCOUNT_ADDRESS,
+        private_key=PREDEPLOYED_ACCOUNT_PRIVATE_KEY,
+    )
+    invoke_block = get_block(parse=True)
+    assert invoke_block["status"] == "ACCEPTED_ON_L2"
+    assert_transaction(invoke_tx_hash, "ACCEPTED_ON_L2")
+
+    # Blocks should be aborted and transactions should be rejected
+    response = abort_blocks(contract_deploy_block["block_hash"])
+    assert response.status_code == 200
+    assert response.json()["aborted"] == [
+        contract_deploy_block["block_hash"],
+        invoke_block["block_hash"],
+    ]
+    last_block = get_block(parse=True)
+    assert last_block["status"] == "ACCEPTED_ON_L2"
+    assert last_block["block_number"] == 0
+    contract_deploy_block_after_abort = get_block(
+        block_hash=contract_deploy_block["block_hash"], parse=True
+    )
+    assert contract_deploy_block_after_abort["status"] == "ABORTED"
+    assert_transaction(contract_deploy_info["tx_hash"], "REJECTED")
+    invoke_block_after_abort = get_block(
```

Comment:
For better readability, I think it would be useful to put newlines after groups of assertion; i.e. to visually separate deploy-block and invoke-block assertion.

Reply:
Fixed in a2be84291710ff1cce0287165dc7cab28a018351